nomad/structs/services.go

```diff
@@ -825,6 +834,10 @@ OUTER:
 		return false
 	}
 
+	if !reflect.DeepEqual(s.TaggedAddresses, o.TaggedAddresses) {
```

Comment:
Could we use `helper.CompareMapStringString` here rather than reflect to avoid the need for type assertion?